### PR TITLE
Add warning links to preview screens for TI and Y!

### PR DESCRIPTION
**Pull request by jsom**

### CODE REVIEW

**Comment by guillaumedebavelaere** [comment]
Do we have a dummy account to test ?

**Comment by guillaumedebavelaere** [comment]
This is not part of this PR, but I noticed that the place order button was clickable even if the acknowledgment were not switched on. Is this something we changed recently ?

**Comment by jsom** [comment]
@guillaumedebavelaere no dummy account yet. Backend work still to be done.

Hmm, no that sounds like a bug. Can you please file a bug?

**Comment by guillaumedebavelaere** [comment]
I added a bug. Otherwise the code looks good.

File: TradeItIosTicketSDK2/TradeItTradePreviewViewController.swift [hunk 1]
```diff
@@ -175,9 +203,8 @@ class TradeItTradePreviewViewController: TradeItViewController, UITableViewDeleg
         var cells: [PreviewCellData] = []
 
         cells += generateWarningCellData()
-
-        acknowledgementCellData = generateAcknowledgementCellData()
```
Comment (guillaumedebavelaere) [inline TradeItIosTicketSDK2/TradeItTradePreviewViewController.swift]: Actually I think it is because of this change

**Comment by jsom** [comment]
This is waiting on more requirements gathering

File: TradeItIosTicketSDK2/TradeItTradePreviewViewController.swift [hunk 2]
```diff
@@ -201,20 +222,16 @@ class TradeItTradePreviewViewController: TradeItViewController, UITableViewDeleg
...
-    private func generateAcknowledgementCellData() -> [AcknowledgementCellData] {
-        guard let acknowledgements = previewOrderResult?.ackWarningsList as? [String] else { return [] }
-
-        return acknowledgements.map({ acknowledgement in
-            return AcknowledgementCellData(acknowledgement: acknowledgement)
-        })
+    private func generateMessageCellData() -> [PreviewCellData] {
+        guard let messages = previewOrderResult?.messages else { return [] }
+        var cellData: [PreviewCellData] = []
+        messages.filter { $0.links.count > 0 }.forEach { message in
```
Comment (jsom) [inline TradeItIosTicketSDK2/TradeItTradePreviewViewController.swift]: Remove the `filter` - just for testing

File: TradeItIosEmsApi/TradeItPreviewMessageLink.h [hunk 3]
```diff
@@ -0,0 +1,12 @@
+#import <JSONModel/JSONModel.h>
+
+@protocol TradeItPreviewMessageLink
+
+@end
+
+@interface TradeItPreviewMessageLink : JSONModel
+
+@property (nonatomic, copy, nonnull) NSString *label;
```
Comment (jsom) [inline TradeItIosEmsApi/TradeItPreviewMessageLink.h]: This apparently should be nullable?

File: TradeItIosTicketSDK2/TradeItYahooPreviewOrderAcknowledgementTableViewCell.swift [hunk 4]
```diff
@@ -6,17 +6,17 @@ class TradeItYahooPreviewOrderAcknowledgementTableViewCell: UITableViewCell {
...
     
-    var cellData: AcknowledgementCellData?
-    internal weak var delegate: AcknowledgementDelegate?
-    
-    func populate(withCellData cellData: AcknowledgementCellData, andDelegate delegate: AcknowledgementDelegate) {
-        self.cellData = cellData
-        self.acknowledgementLabel.text = cellData.acknowledgement
-        self.delegate = delegate
-    }
+    var cellData: PreviewCellData?//AcknowledgementCellData?
```
Comment (jsom) [inline TradeItIosTicketSDK2/TradeItYahooPreviewOrderAcknowledgementTableViewCell.swift]: Remove

File: TradeItIosEmsApi/TradeItPreviewMessageLink.h (hunk 3 above)
Comment (jsom) [reply TradeItIosEmsApi/TradeItPreviewMessageLink.h]: Ignore. Now comes back with default from server.

**Comment by guillaumedebavelaere** [comment]
@jsom the code looks good. 
However, I tested with Dummy and there is a small UI issue: the checkbox in the last acknowledgment cell doesn't render correctly (the grey circle doesn't appear), see attached screenshot.

screen shot 2017-10-09 at 4 36 23 pm

**Comment by jsom** [comment]
@guillaumedebavelaere fixed

**Pull request merged**